wdl-engine/src/backend/crankshaft.rs

```diff
@@ -471,4 +486,154 @@ impl TaskExecutionBackend for CrankshaftBackend {
+            return Ok(rx);
+        }
+
+        let inner_backend = self.inner.clone();
+        let generator = self.generator.clone();
+        let output_path = output_dir.to_path_buf();
+
+        tokio::spawn(async move {
+            let result = async {
+                let (uid, gid) = get_uid_gid();
+                let ownership = format!("{uid}:{gid}");
+                info!(
+                    "cleanup target: '{}', attempting to set ownership to: {}",
+                    output_path.display(),
+                    ownership
+                );
+
+                if !output_path.exists() {
+                    info!("output directory does not exist, skipping cleanup");
+                    return Ok(());
+                }
+                if !output_path.is_dir() {
+                    bail!(
+                        "output directory `{path}` is not a directory",
+                        path = output_path.display()
+                    );
+                }
+
+                let output_mount = Input::builder()
+                    .path(GUEST_OUT_DIR)
+                    .contents(Contents::Path(output_path.clone()))
+                    .ty(Type::Directory)
+                    // need write access
+                    .read_only(false)
+                    .build();
+
+                let cleanup_task_name = format!(
+                    "wdl-engine-chown-cleanup-{}",
+                    generator
+                        .lock()
+                        .expect("generator should always acquire")
+                        .next()
+                        .expect("generator should never be exhausted")
+                );
+
+                let cleanup_resources = Resources::builder()
+                    .cpu(CLEANUP_CPU)
+                    .ram(CLEANUP_MEMORY)
+                    .build();
+
+                let cleanup_task = Task::builder()
+                    .name(&cleanup_task_name)
+                    .executions(NonEmpty::new(
+                        Execution::builder()
+                            .image("alpine:latest")
+                            .program("chown")
+                            .args([
+                                "-R".to_string(),
+                                ownership.clone(),
+                                GUEST_OUT_DIR.to_string(),
+                            ])
+                            .work_dir("/")
+                            .build(),
+                    ))
+                    .inputs([Arc::new(output_mount)])
+                    .resources(cleanup_resources)
+                    .build();
+
+                info!(
+                    "running cleanup task '{}' to chown '{}' to '{}'",
+                    cleanup_task_name,
+                    output_path.display(),
+                    ownership
+                );
+
+                let (spawned_tx, _) = oneshot::channel();
+                let token = CancellationToken::new();
```

Comment:
We should probably be passing in the same cancellation token that we're using for evaluation rather than this new one, as there would be no way for the caller to signal the token as cancelled. 

That is to say we should pass the token in as an argument to this function.